### ISSUE TITLE
Holodex+ breaks channel icon and subscribe button

### ISSUE THREAD

**Issue by Robot-Inventor** [HolodexNet/Holodex-Plus#17]
When Holodex+ extension is enabled, channel icon and subscribe button in embedded player are broken.

### When Holodex+ is disabled

![image](https://user-images.githubusercontent.com/50550561/191053018-376375d9-5fd8-47e3-92a4-db3b9d41351f.png)

### When Holodex+ is enabled

![image](https://user-images.githubusercontent.com/50550561/191052718-91bd1928-a717-4f70-b921-01d09ec943b5.png)

## Device

- Firefox Developer Edition v105.0b9
- Windows 11 21H2
- Holodex+ v0.0.4

**Comment by sphinxrave** [comment]
Ah, thanks for the bug report... it's probably some of the flags we override for the YT Player here https://github.com/HolodexNet/Holodex-Plus/blob/dev/src/content/yt-player-overrides.inject.ts , but will have to investigate a bit more...

**Comment by Robot-Inventor** [comment]
@sphinxrave Thanks for the reply. I'll share some info what I investigated and found out.

- Holodex+ for Chrome doesn't break the channel icons. The problem only occurs when using Firefox
- Firefox's Enhanced Tracking Protection status has nothing to do with this issue.
- It seems that if the ".ytp-title-enable-channel-logo" class is set to "#movie_player" element in the embedded player, it overrides the "display: none;" of the channel icon and displays it.
![image](https://user-images.githubusercontent.com/50550561/191291540-ec03a32b-2dd2-4468-8b08-582e4e6048b1.png)
- The ".ytp-title-enable-channel-logo" class is not set when Holodex+ is enabled in Firefox

If I find out anything new, I'll share it with you again😃

**Comment by sphinxrave** [comment]
... Actually i can't reproduce it

![image](https://user-images.githubusercontent.com/62570796/191392692-b3f1296e-eee3-45a8-8938-826218ef9ec4.png)

**Comment by Robot-Inventor** [comment]
...I have to investigate more. Some browser settings and Holodex+ may be interfering.

**Comment by sphinxrave** [comment]
Yeah, but maybe don't spend so much time on it considering it's just a minor UI issue that isn't broadly affecting people 👍 

It's more important that Holodex+ should add a feature for identifying when 3rd Party Cookies are disabled and YT within iframe doesn't know who the user is, then tell Holodex by `window.postMessage` to alert the user to fix it.

**Comment by Robot-Inventor** [comment]
I checked again today and the channel icons were showing up. However, sometimes they don't show up...

https://user-images.githubusercontent.com/50550561/191537576-7d30189f-4b98-4ba3-8264-7f080e79d9ec.mp4

But I agree with you that this issue is not a big one and that 3rd party cookie related features are more important.🙂 I had a hard time finding the extension that was causing log in issue, and some people on Twitter were saying they have same one.

Especially baffling with DuckDuckGo Privacy Essentials, adding Holodex and YouTube to the whitelist still causes problems with video playback☹️

**Comment by sphinxrave** [comment]
It should be possible to detect the third party cookie issue, but i wonder how to word the 'popup'. It's such a technical problem that I have a hard time conveying it simply to an average user. "Hey just change your browser's privacy settings and validate you don't have any extensions that impact 'third party cookies'" is very wordy and complicated 😢

**Comment by Robot-Inventor** [comment]
Choosing a message to display to a user is always a difficult problem. 

I think the simplest message is to specify the name of extension like "Please disable Privacy Badger. It may occur some problems." But can Holodex identify installed extensions?

You may be able to identify the extension by examining the code inserted on the page, but it is technically difficult. 🤔

**Comment by sphinxrave** [comment]
Doesn't survive updates, also would be the most powerful anti-adblock detection script i've ever seen.

**Comment by Robot-Inventor** [comment]
Hmm... Well then, how about the message like, "Please disable the extensions on following list", and list major extensions that may cause problems regardless of whether they are installed by the user?
